### ISSUE TITLE
OSC 6.0 + PagSeguro 2.4

### ISSUE THREAD

**Issue by VIPete** [deivisonarthur/OSC-Magento-Brasil-6-Pro#80]
Olá!

(perdoa-me quando escrever Português errado, sou Holandês)

Estou usando o OSC 6 mas quando tentar usar PagSeguro 2.4 para magento ele não passa para página de pagseguro. Da erro e mostra carrinho vazio.

O log diz: 

2015-03-11T14:57:44+00:00 DEBUG (7): Invalid method UOL_PagSeguro_Model_PaymentMethod::_redirectUrl(Array
(
    [0] => http://universalbeleza.com.br/novo/checkout/onepage
)
)
2015-03-11T14:57:44+00:00 ERR (3): Notice: Undefined variable: checkout  in /home/universa/public_html/novo/app/code/community/UOL/PagSeguro/controllers/PaymentController.php on line 203

No payment controller tem indicações de versão antigo de OSC 
https://github.com/pagseguro/magento/blob/master/app/code/community/UOL/PagSeguro/controllers/PaymentController.php

Line 77:
 $fileOSC = scandir(getcwd().'/app/code/community/DeivisonArthur');

Mas não funçiona com versão novo que fica na pasta /Inovarti 

Algum idea?

**Comment by rafaelstz** [comment]
Tenta colocar os arquivos do OSC relativos ao tema, também em seu tema app/design e skin/.
Verifique as configurações do OSC para ver se está ativo.
Depois de instalar reindexe e limpe o cachê.
Sem o PagSeguro ele funciona?

**Comment by VIPete** [comment]
Sim, sem PagSeguro funciona bem, Os error no log também sempre são ligado com PagSeguro. Postei no forum do PagSeguro também, mas tenho mais confiança das pessoas aqui. :) 

Eu acho que o problema fica no app/code/community/UOL/PagSeguro/controllers/PaymentController.php 

Eu poderia alterar esse arquivo para funcionar, mas estou buscando como que faço...

**Comment by rafaelstz** [comment]
Bom, vi que o tem caminhos errados que provavelmente são do OSC antigo, mas instalei esse mesmo módulo do PagSeguro e funcionou corretamente há um mês. O PagSeguro funciona sem o OSC?

**Comment by VIPete** [comment]
Sim, mas também estou usando magento 1.9.1.0 agora. Qual versão do magento você usa?

**Comment by rafaelstz** [comment]
1.9.1.0

**Comment by isaaclopes** [comment]
Ola @VIPete tenta dessa forma para versao OSC 6 instalada

$fileOSC = scandir(getcwd().'/app/code/community/Inovarti/Onestepcheckout');
if ($fileOSC) {
 $enabledOSC = Mage::helper('onestepcheckout/config')->isEnabled();
}
$feedback = ($enabledOSC == false ? 'checkout/onepage' : 'onestepcheckout');


em:

https://github.com/pagseguro/magento/blob/master/app/code/community/UOL/PagSeguro/controllers/PaymentController.php#L77

**Comment by bnubruno** [comment]
tenho exatamente o mesmo problema, achou a solução?

**Comment by bnubruno** [comment]
Fiz o que é @isaaclopes disse, e agora não da mais o erro. Porém após a finalização da compra, o carrinho limpa e nada acontece.

**Comment by rafaelstz** [comment]
Aparece algo no arquivo de **log** ?

**Comment by bnubruno** [comment]
Depois que fiz a alteração não. Mas resolvi utilizar o módulo do pagseguro transparente e funcionou. Já quebra o galho. Obrigado.
